### ISSUE TITLE
Inaccurate potential, maybe due to lost imaginary part

### ISSUE THREAD

**Issue by ErikMarklund** [maroba/multipoles#18]
**Describe the bug**
Some simple charge distribusions give inaccurate results. For example, placing two opposite charges along the y-axis (0, +-1, 0) gives zero coefficients for the l=1 multipoles, and the electrostatic potential is zero at (0,2,0), which is wrong.

I suspect that the reason is that the code throws away the imaginary part of the multipoles:

```
    def _eval(self, xyz, l_max=None):
        ...
        return np.array(
                [sum(self._multipole_contribs((x, y, z))[:l_max + 1]) for x, y, z in zip(*xyz)]
            )

...

    def _multipole_contribs(self, xyz):
    ...
        mp_contribs = []

        for l in range(self.l_max + 1):
            phi_l = 0
            for m in range(-l, l + 1):
                Y_lm = sph_harm(m, l, phi, theta)
                q_lm = self.multipole_moments[(l, m)]
                if self.exterior:
                    phi_l += np.sqrt(4 * np.pi / (2 * l + 1)) * q_lm * Y_lm / r ** (l + 1)
                else:
                    phi_l += np.sqrt(4 * np.pi / (2 * l + 1)) * q_lm * Y_lm * r ** l
            mp_contribs.append(phi_l.real)
        return mp_contribs
```
For charges along the y-axis, the real part of Y_1{-1,0,1} becomes zero, so they contribute nothing to the potential.

**To Reproduce**
```
#!/usr/bin/env python

from math import *
from multipoles import MultipoleExpansion


def print_moments(Phi):
    moments = Phi.multipole_moments
    for k,v in moments.items():
        print(f'l={k[0]:1}, m={k[1]:2}: {v}')

if __name__=='__main__':
    
    # exp(iz) = cos(z) + i sin(z)
    # Y-11 = ... sin(theta)exp(-iphi)
    #      = ... sin(theta)(cos(-phi) + isin(-phi))
    # Y01  = ... cos(theta)
    # Y11  = ... sin(theta)exp(iphi)
    # So theta=pi/2 and phi=pi/2 should give a zero real component for all.
    
    qd = {
        'discrete': True,
        'charges': [{'q': 1, 'xyz': (0,1,0)}, {'q': -1, 'xyz': (0,-1,0)}]
        }
    
    l_max = 1
    Phi = MultipoleExpansion(qd, l_max)
    print_moments(Phi)

    print(f'Epot(0,2,0) = {Phi(0,2,0)}')
```

The output is:
```
l=0, m= 0: 0.0
l=1, m=-1: 0.0
l=1, m= 0: 0.0
l=1, m= 1: 0.0
Epot(0,2,0) = 0.0
```

**Expected behavior**
The electrostatic potential should be positive (0.5 I think).

**Desktop (please complete the following information):**
 - OS: [e.g. macOS]
 - Version [e.g. 0.3.2]

**Additional context**
Probably/hopefully irrelevant, but I have not installed using pip. Instead I made a clone and a symlink to the multipoles folder. Allows me to fiddle with the code and try things out (which I haven't done yet so everything in the report is with the unmodified code).

Disclaimer: Not an expert at multipole expansions, but throwing away the imaginary part seems wrong.

**Comment by ErikMarklund** [comment]
I implemented a simple fix. Seems to work for the test case of a two-point-charge dipole. If you have some standard tests I can unleash it on I can do that too to make sure I didn't break something else.

Below are three consecutive runs for dipoles in the x, y, and z directions.
```
{'discrete': True, 'charges': [{'q': 1, 'xyz': (1, 0, 0)}, {'q': -1, 'xyz': (-1, 0, 0)}]}
l=0, m= 0: 0j
l=1, m=-1: (1.4142135623730951-8.659560562354933e-17j)
l=1, m= 0: 0j
l=1, m= 1: (-1.4142135623730951-8.659560562354933e-17j)
Epot(2,0,0) = (0.5000000000000001+0j)

{'discrete': True, 'charges': [{'q': 1, 'xyz': (0, 1, 0)}, {'q': -1, 'xyz': (0, -1, 0)}]}
l=0, m= 0: 0j
l=1, m=-1: 1.4142135623730951j
l=1, m= 0: 0j
l=1, m= 1: 1.4142135623730951j
Epot(0,2,0) = (0.5000000000000001+0j)

{'discrete': True, 'charges': [{'q': 1, 'xyz': (0, 0, 1)}, {'q': -1, 'xyz': (0, 0, -1)}]}
l=0, m= 0: 0j
l=1, m=-1: 0j
l=1, m= 0: (2+0j)
l=1, m= 1: 0j
Epot(0,0,2) = (0.5+0j)
```

As you can see, the potential is no longer dependent on how the coordinate system is rotated (at least not in this case).

**Comment by ErikMarklund** [comment]
... if this works I make a pull request with the fix.

**Comment by ErikMarklund** [comment]
Pushed my commits. See complex branch in my fork https://github.com/ErikMarklund/multipoles/tree/complex

**Comment by maroba** [comment]
Hi, sorry for the late reply, and thanks for investigating this bug. In fact, I never used the package for discrete charge distributions for production purpose, so it is not too surprising that it still contained bugs. If you like, send me a pull request so that I can feed it back into the repo and you become a contributor. Thanks again!

**Comment by maroba** [comment]
@ErikMarklund Ok, I had time to check. Indeed, discarding the imaginary part of the multipole moments q_lm in the discrete case is nonsense, of course. However, discarding the imaginary part of the multipole contributions of angular moment $l$ to the potential makes sense, I think, because the sum over the contributions over $m$ for fixed $l$ should yield a real contribution to the potential. But for numerical reasons, there will always be a tiny imaginary part. 

I have uploaded version 0.4.1 with the fixes to PyPi.

**Comment by ErikMarklund** [comment]
Hi, sorry I too missed the conversation, for a whole year! Not sure how that happened because I remember checking this thread a couple of times. Does it yield sensible results for simple test systems like the one I had?

Btw I used the code to calculate the potential around zwitterionic macromolecules, where different residues can be either neutral +1 or -1, so discrete distributions were the way to go and I got some unexpected results. That's why I ran the tests for simple things.
